### PR TITLE
Refactor bearer token gRPC classes

### PR DESCRIPTION
**Pull request by tiziano88**
Remove obsolete application protobuf and functions. The remote
attestation gRPC service is no longer used anywhere, as we have now
generally agreed that information flow is the way that labels are
propagated and enforced in Oak.

Ref #488

### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/client/application_client.h [hunk 1]
```diff
@@ -48,21 +48,33 @@ const std::string kBearerTokenHmacData{"oak-grpc-bearer-token-1"};
...
+        oak::utils::hmac_sha256(authorization_bearer_token, kBearerTokenHmacData).ValueOrDie();
+
+    return grpc::CompositeCallCredentials(
+        grpc::MetadataCredentialsFromPlugin(
+            absl::make_unique<AuthorizationBearerTokenMetadata>(authorization_bearer_token)),
+        grpc::MetadataCredentialsFromPlugin(absl::make_unique<PolicyMetadata>(
+            AuthorizationBearerTokenPolicy(authorization_bearer_token_hmac))));
+  }
+
+  // Generates a fresh nonce to use as authorization bearer token, and use that to authenticate and
```
Comment (conradgrobler) [inline oak/client/application_client.h]: nit: s/use/uses/

File: oak/client/application_client.h [hunk 2]
```diff
@@ -90,25 +102,11 @@ class ApplicationClient {
   // on additional metadata or specific time-related policies.
   //
   // See https://grpc.io/docs/guides/auth/.
-  static std::shared_ptr<grpc::Channel> CreateChannel(std::string addr) {
+  static std::shared_ptr<grpc::Channel> CreateChannel(
+      std::string addr, std::shared_ptr<grpc::CallCredentials> call_credentials) {
```
Comment (daviddrysdale) [inline oak/client/application_client.h]: drive-by thought: Why not create a `CreateChannel(addr)` overload that forwards on to `CreateChannel(addr, private_authorization_bearer_token_call_credentials())`?  That would reduce the churn on `examples/*/client/*cc`.
